### PR TITLE
add check_mode for elb_application_lb* & refactor integration tests

### PR DESCRIPTION
**Pull request by jatorcasso**
##### SUMMARY
Add check_mode support for elb_application_lb* & refactor integration tests.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
elb_application_lb
elb_application_lb_info

### CODE REVIEW

File: tests/integration/targets/elb_application_lb/tasks/main.yml
Comment (alinabuzachis) [inline tests/integration/targets/elb_application_lb/tasks/main.yml:461]: @jillr Do we want to keep all this messaging for the elb? I guess this comes from module_utils.

**Review by alinabuzachis — COMMENTED** [review]
Thanks @jatorcasso. I left some initial comments. I will do a more accurate review.

File: plugins/modules/elb_application_lb.py
Comment (jatorcasso) [reply plugins/modules/elb_application_lb.py:532]: @alinabuzachis I stole that idea from [ec2_vol](https://github.com/ansible-collections/amazon.aws/pull/509/files), but I actually really like the idea of having a message there! Is there a standard way of returning on check_mode?

File: tests/integration/targets/elb_application_lb/tasks/main.yml
Comment (jatorcasso) [reply tests/integration/targets/elb_application_lb/tasks/main.yml]: done!

File: plugins/modules/elb_application_lb.py
Comment (jillr) [reply plugins/modules/elb_application_lb.py:532]: > Is there a standard way of returning on check_mode?

Nope!  :smile: 
In the majority of cases AWS modules will do one of:
* Return a `msg` like you have here
* Return only a `changed` key (based on if there would have been a change, if not in check mode)
* Return something that resembles the normal RETURN of the module (like Alina has noted)

Some of this is just the legacy of 7 or 8 years of organic changes to the AWS modules.  Some of it is what makes the most sense for the way the module code has been structured (since we also don't have a defined standard that we expect for that either).  The Ansible devel guide doesn't dictate a preference for what check_mode should return, and the user guide doesn't promise anything, so it's collection/developer preference.  I'm personally ok with having check_mode return only a `changed` or having a `msg`.

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by jatorcasso** [comment]
recheck

File: tests/integration/targets/elb_application_lb_info/tasks/main.yml
Comment (alinabuzachis) [inline tests/integration/targets/elb_application_lb_info/tasks/main.yml:109]: Can you please organise the tests as tremble suggested in this comment https://github.com/ansible-collections/amazon.aws/pull/509#issuecomment-932056400?

**Review by alinabuzachis — CHANGES_REQUESTED** [review]

File: tests/integration/targets/elb_application_lb_info/tasks/main.yml
Comment (jatorcasso) [reply tests/integration/targets/elb_application_lb_info/tasks/main.yml:109]: yes - wasnt sure if it needed that depth on the info module but I will update

Comment (alinabuzachis) [reply tests/integration/targets/elb_application_lb_info/tasks/main.yml:109]: I mean for the elb_application_lb module, not for the info module (Sorry, pinned the comment on the wrong file).

Comment (alinabuzachis) [reply tests/integration/targets/elb_application_lb_info/tasks/main.yml:109]: So, I actually missed the tests for elb_application_lb.

File: plugins/modules/elb_application_lb.py
Comment (jatorcasso) [reply plugins/modules/elb_application_lb.py:506]: updated! fixed incorrect param type as well.

File: plugins/modules/elb_application_lb.py
Comment (alinabuzachis) [inline plugins/modules/elb_application_lb.py:439]: I guess there is no need to document `resource_actions:`.

**Review by alinabuzachis — APPROVED** [review]
Other than that, it looks good to me. Thank you @jatorcasso. Let's wait for some other review from @marknet15 @markuman.

File: plugins/modules/elb_application_lb.py
Comment (jatorcasso) [reply plugins/modules/elb_application_lb.py:439]: I was just covering every parameter returned.. does this mean I should also remove `changed` and `failed` ?

Comment (marknet15) [reply plugins/modules/elb_application_lb.py:439]: Yep don't think other modules have this added here and I don't think its relevant to the module itself per say.

File: plugins/modules/elb_application_lb_info.py
Comment (marknet15) [inline plugins/modules/elb_application_lb_info.py:92]: I don't think this is applicable for a info module as nothing would change

File: plugins/modules/elb_application_lb_info.py
Comment (marknet15) [inline plugins/modules/elb_application_lb_info.py:172]: Same as on the earlier comment from @alinabuzachis probably no point including this here

File: tests/integration/targets/elb_application_lb/tasks/main.yml
Comment (marknet15) [inline tests/integration/targets/elb_application_lb/tasks/main.yml:1001]: Not sure what others think but I didn't think check mode was really applicable for info modules as nothing really changes

Comment (alinabuzachis) [reply tests/integration/targets/elb_application_lb/tasks/main.yml:1001]: I know that all the modules should support check_mode. You're right, there is no change. _info modules are a particular case, but don't know about the story behind the check_mode here.

**Review by marknet15 — APPROVED** [review]

File: tests/integration/targets/elb_application_lb/tasks/main.yml
Comment (marknet15) [reply tests/integration/targets/elb_application_lb/tasks/main.yml:1001]: Yeah makes sense, guess they are indeed quite a particular case

Comment (jatorcasso) [reply tests/integration/targets/elb_application_lb/tasks/main.yml:1001]: Yeah, I was basing that off a conversation with @jillr who basically said it should be there so nothing breaks if the user does --check on an entire playbook

Comment (marknet15) [reply tests/integration/targets/elb_application_lb/tasks/main.yml:1001]: Ahhh cool cool 👍🏻

**Review by jillr — CHANGES_REQUESTED** [review]
@jatorcasso Can you please retain the ALB naming in this set of modules and tests?  ELBs are generally understood to mean v1/Classic Load Balancers (covered by amazon.aws.elb_classic_lb, the module formerly known as ec2_elb_lb).  Only Application Load Balancers (elbv2) are covered by these modules.
https://docs.aws.amazon.com/elasticloadbalancing/index.html
The term ALB is a widely-used and understood convention for AWS users https://cloudacademy.com/blog/application-load-balancer-vs-classic-load-balancer/

**Comment by jatorcasso** [comment]
> @jatorcasso Can you please retain the ALB naming in this set of modules and tests? ELBs are generally understood to mean v1/Classic Load Balancers (covered by amazon.aws.elb_classic_lb, the module formerly known as ec2_elb_lb). Only Application Load Balancers (elbv2) are covered by these modules. https://docs.aws.amazon.com/elasticloadbalancing/index.html The term ALB is a widely-used and understood convention for AWS users https://cloudacademy.com/blog/application-load-balancer-vs-classic-load-balancer/

@jillr changed to ALB where it made the most sense to me. Please let me know if I missed any, thanks!

**Pull request merged**
